linux_os/guide/system/network/network-firewalld/ruleset_modifications/ensure_firewall_rules_for_open_ports/oval/shared.xml

```diff
@@ -1,76 +1,183 @@
+          <criterion comment="Check firewall rules for tcp listening ports"
+            test_ref="test_listening_fw_inet_ports_tcp_test" />
+        </criteria>
+      </criteria>
+      <criteria operator="OR">
+        <criterion comment="Check if any service is listening on udp"
+          test_ref="test_listening_inet_ports_udp_exist" />
+        <criteria operator="AND">
+          <criterion comment="Check there are defined udp ports in firewall"
+            test_ref="test_var_firewalled_udp_ports_exists" />
+          <criterion comment="Check firewall rules for udp listening ports"
+            test_ref="test_listening_fw_inet_ports_udp_test" />
+        </criteria>
+      </criteria>
     </criteria>
   </definition>
 
-  <local_variable id="var_firewalled_ports" version="1" comment="Firewalled ports according to firewalld configuration" datatype="int">
-    <object_component object_ref="object_firewalled_service_port" item_field="subexpression"/>
+<!-- States describing listening objects on loopback interface or established connections-->
+  <linux:inetlisteningservers_state id="state_ipv4_loopback_listening_inet_ports" version="1"
+    comment="Checks local address is not ipv4 loopback."
+    xmlns="http://oval.mitre.org/XMLSchema/oval-definitions-5#linux">
+    <linux:protocol operation="pattern match">^.*$</linux:protocol>
+    <linux:local_address operation="equals">127.0.0.1</linux:local_address>
+    <linux:local_port datatype="int" operation="greater than or equal">0</linux:local_port>
+  </linux:inetlisteningservers_state>
+
+  <linux:inetlisteningservers_state id="state_ipv6_loopback_listening_inet_ports" version="1"
+    comment="Checks local address is not ipv6 loopback."
+    xmlns="http://oval.mitre.org/XMLSchema/oval-definitions-5#linux">
+    <linux:protocol operation="pattern match">^.*$</linux:protocol>
+    <linux:local_address operation="equals">::1</linux:local_address>
+    <linux:local_port datatype="int" operation="greater than or equal">0</linux:local_port>
+  </linux:inetlisteningservers_state>
+
+  <linux:inetlisteningservers_state id="state_inet_foreign_port_connected" version="1"
+    comment="Checks this is a listening service not connected">
+    <linux:foreign_port datatype="int" operation="not equal">0</linux:foreign_port>
+  </linux:inetlisteningservers_state>
+
+  <!-- Objects and variables describing the firewalld configuration -->
+  <ind:textfilecontent54_object id="object_active_firewalld_zone_cfgs" version="1">
+    <ind:path>/etc/firewalld/zones/</ind:path>
+    <ind:filename operation="pattern match">^.+\.xml$</ind:filename>
+    <ind:pattern operation="pattern match">^[\s]*&lt;interface\s*name=.*$</ind:pattern>
+    <ind:instance operation="greater than or equal" datatype="int">1</ind:instance>
+  </ind:textfilecontent54_object>
+
+  <local_variable id="var_firewalld_active_zones" version="1"
+    comment="Firewalled ports according to firewalld configuration" datatype="string">
```

Comment:
This variable actually includes the filenames of firewalld active zones instead of firewalled ports.